src/ConstructiveSolidGeometry/VolumePrimitives/Box.jl

```diff
@@ -1,84 +1,105 @@
+    end
+    scale = SVector{3, T}(1, 1, 1)
+    if haskey(dict, "scale")
+        # scale = parse_translate_vector(T, dict["translate"], length_unit)
+        @info "Scale not yet implemented"
+    end
+    rot = if haskey(dict, "rotation")
+        parse_rotation_matrix(T, dict["rotate"], length_unit)
+    else
+        one(SMatrix{3, 3, T, 9})
+    end
+    hX = typeof(x) <: Real ? x : width(x)/2
+    hY = typeof(y) <: Real ? y : width(y)/2
+    hZ = typeof(z) <: Real ? z : width(z)/2
+    return Box{T, ClosedPrimitive}(
+        scale[1] * hX, 
+        scale[2] * hY, 
+        scale[3] * hZ, 
+        origin, rot
+    )
 end
 
-function Dictionary(b::Box{T}) where {T}
-    dict = OrderedDict{String,Any}()
-    dict["x"] = typeof(b.x) == T ? b.x * 2 : OrderedDict{String,Any}("from" => b.x.left, "to" => b.x.right)
-    dict["y"] = typeof(b.y) == T ? b.y * 2 : OrderedDict{String,Any}("from" => b.y.left, "to" => b.y.right)
-    dict["z"] = typeof(b.z) == T ? b.z * 2 : OrderedDict{String,Any}("from" => b.z.left, "to" => b.z.right)
-    OrderedDict{String,Any}("box" => dict)
+function vertices(b::Box{T}) where {T}
+    return SVector{8, CartesianPoint{T}}(
+        b.rotation * SVector{3,T}(-b.hX, -b.hY, -b.hZ) .+ b.origin,
+        b.rotation * SVector{3,T}(+b.hX, -b.hY, -b.hZ) .+ b.origin,
+        b.rotation * SVector{3,T}(+b.hX, +b.hY, -b.hZ) .+ b.origin,
+        b.rotation * SVector{3,T}(-b.hX, +b.hY, -b.hZ) .+ b.origin,
+        b.rotation * SVector{3,T}(-b.hX, -b.hY, +b.hZ) .+ b.origin,
+        b.rotation * SVector{3,T}(+b.hX, -b.hY, +b.hZ) .+ b.origin,
+        b.rotation * SVector{3,T}(+b.hX, +b.hY, +b.hZ) .+ b.origin,
+        b.rotation * SVector{3,T}(-b.hX, +b.hY, +b.hZ) .+ b.origin,
+    )
 end
 
-get_x_limits(b::Box{T}) where {T} = (_left_linear_interval(b.x), _right_linear_interval(b.x))
-get_y_limits(b::Box{T}) where {T} = (_left_linear_interval(b.y), _right_linear_interval(b.y))
-get_z_limits(b::Box{T}) where {T} = (_left_linear_interval(b.z), _right_linear_interval(b.z))
+sample(b::Box) = vertices(b)
 
-function get_decomposed_surfaces(b::Box{T}) where {T}
-    xMin::T, xMax::T = get_x_limits(b)
-    yMin::T, yMax::T = get_y_limits(b)
-    zMin::T, zMax::T = get_z_limits(b)
-    tol = geom_atol_zero(T)
-    if isapprox(xMin, xMax, atol = tol)
-        return AbstractSurfacePrimitive[RectangleX(b, xMin)]
-    elseif isapprox(yMin, yMax, atol = tol)
-        return AbstractSurfacePrimitive[RectangleY(b, yMin)]
-    elseif isapprox(zMin, zMax, atol = tol)
-        return AbstractSurfacePrimitive[RectangleZ(b, zMin)]
-    else
-        return AbstractSurfacePrimitive[
-                                            RectangleX(b, xMin), RectangleX(b, xMax), 
-                                            RectangleY(b, yMin), RectangleY(b, yMax), 
-                                            RectangleZ(b, zMin), RectangleZ(b, zMax)
-                                        ]
-    end
+function faces(b::Box{T}) where {T}
```

Comment:
While this is type stable, and returns a vector of constant length, this will not be possible for the cone primitive (which covers cylinders and cones). For example a simple cylinder has 3 surfaces, while a simple cone only has 2. 
Similar cases arise for other volume primitives.

Reply:
Our previous `Cone` was a parametric type which included the information how many surfaces it has. 
E.g. a cone with `(r_min_top = 0.5, r_max_top = 1.0, r_min_bot, 0.2, r_max_bot = 1.2)` (and full in phi) will always return 
4 surfaces. Also the types of the 4 surfaces are fixed.

Reply:
Note that as written you can't use the parametric type to differentiate this, because the 2 surfaces case applies to cones with `rtop = 0` or `rbot = 0`

Reply:
```julia
struct Cone{T, RT,...}
  r_top::RT
  ...
end
```
for `rtop = 0` `RT = T` whearas for `rtop != 0` `RT` will be for example `RT = Tuple{T,T}`

Reply:
To be clearer `Cone( T, Tuple{T,T}, Nothing, T)` has 2 surfaces if `rtop = 0` or `rbot = 0` and 3 surfaces otherwise. Therefore the parametric type does not work to differentiate this.

Reply:
So this is a suggested change for the `Cone` correct?

Reply:
Well, this could be solved if we only allow for full `Cones` define a `Cone(r = 1..2)` as `CSGDifference(Cone(2), Cone(1))`.

Reply:
Ah, you ment also `r_max == r_min == 0` for either top or bot, right?
Than `r_top` could be set to `nothing` instead of `0`. By this you could dispatch appropriately.

Reply:
Yes exactly. My point is that we will need to change the `Cone` so we can do this

Reply:
And yes now would be also a good time to decide if we want to restrict to only full cones as @fhagemann suggests. This decision would also propagate to Tori and others where we can use CSGDifference

Reply:
Yes, we have to. 

I would suggest to define it like this:
```julia
struct Cone{T, CO, RT, RB} <: AbstractVolumePrimitive{T, CO}
    r_top::RT
    r_bot::RB
    hZ::T
    origin::CartesianPoint{T}
    rotation::SMatrix{3,3,T,9}
end
```
But we can first start to implement just the methods of all the functions we need  
for the simple case without the borehole and `r_top` and `r_bot` both larger than 0. 
So for `Cone{T, CO, <:Real, <:Real}`.